ArmPkg/Drivers/SmmuDxe/SmmuDxe.c

```diff
@@ -623,8 +597,8 @@ SmmuV3Configure (
     goto End;
   }
 
-  // Allocate Linear Stream Table
-  StreamTablePtr = SmmuV3AllocateStreamTable (SmmuInfo, &StreamTableLog2Size, &StreamTableSize);
+  TwoLevelStreamTable = (SmmuInfo->StreamTableEntryMax >= (EFI_PAGE_SIZE / sizeof (SMMUV3_STREAM_TABLE_ENTRY)));

```

Comment:
My assumption is this is an optimization no not waste a second page in the case that a single page would suffice or is this explicitly not allowed. Do you know if it is common for us to have less than a page worths of entries?